### PR TITLE
Added ability to change timeout value for Groovy Scripting. Possibili…

### PR DESCRIPTION
**Pull request by viktor-yengovatov**
…ty to choose TLS version for SSL connections: v1, v1.1 and v1.2

Signed-off-by: Viktors Jengovatovs <viktors.jengovatovs@pearlconsulting.no>

Signed-off-by: Your Real Name your.email@email.com

### CODE REVIEW

File: src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java [hunk 1]
```diff
@@ -111,22 +111,50 @@ public String login(@NotNull Project project, @NotNull HybrisRemoteConnectionSet
...
     @NotNull
-    public final HttpResponse post(@NotNull Project project, @NotNull String actionUrl, @NotNull List<BasicNameValuePair> params, boolean canReLoginIfNeeded) {
+    public final HttpResponse post(
+        @NotNull Project project,
+        @NotNull String actionUrl,
+        @NotNull List<BasicNameValuePair> params,
+        boolean canReLoginIfNeeded,
+        final long timeout
+    ) {
+        return post(project, actionUrl, params, canReLoginIfNeeded, 6000L);
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java]: I think here is the recursion in 'post' method

File: src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java [hunk 2]
```diff
@@ -111,22 +111,50 @@ public String login(@NotNull Project project, @NotNull HybrisRemoteConnectionSet
...
-        String csrfToken = getCsrfToken(getHostHacURL(project), sessionId);
+        String csrfToken = getCsrfToken(getHostHacURL(project), sessionId, project, settings);
         if (csrfToken == null) {
             this.sessionId = null;
             if (canReLoginIfNeeded) {
                 return post(project, actionUrl, params, false);
             }
             return createErrorResponse("Unable to obtain csrfToken for sessionId="+sessionId);
         }
-        HttpClient client = createAllowAllClient(6000L);
```
Comment (MykytaKostiuk) [inline src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java:129]: If we have 6 sec as default - please make them as constant and min start value in the spinner.

File: src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java [hunk 3]
```diff
@@ -111,22 +111,50 @@ public String login(@NotNull Project project, @NotNull HybrisRemoteConnectionSet
...
+    }
+
+    @NotNull
+    public final HttpResponse post(
+        @NotNull Project project,
+        @NotNull String actionUrl,
+        @NotNull List<BasicNameValuePair> params,
+        boolean canReLoginIfNeeded
+    ) {
+        return post(project, actionUrl, params, canReLoginIfNeeded, 6000L, null);
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java]: If we have 6 sec as default - please make it a constant and as a start value in the UI spinner.

File: src/com/intellij/idea/plugin/hybris/toolwindow/RemoteConnectionDialog.java [hunk 4]
```diff
@@ -41,14 +41,17 @@
     private JTextField displayNameTextField;
     private OnOffButton sslButton;
     private JLabel sslLabel;
+    private JComboBox sslProtocol;
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/toolwindow/RemoteConnectionDialog.java:44]: If the SSL toggle is off we could disable sslProtocol UI configuration.

File: src/com/intellij/idea/plugin/hybris/project/configurators/impl/DefaultLibRootsConfigurator.java [hunk 5]
```diff
@@ -101,6 +101,7 @@ public void configure(
             hybrisModuleDescriptor.createBootstrapLib(sourceCodeRoot, modifiableModelsProvider);
         }
 
+        System.out.println(">>>" + moduleDescriptor.getClass() + " " + moduleDescriptor.getName());
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/project/configurators/impl/DefaultLibRootsConfigurator.java]: Is it just for debugging?

File: src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java (hunk 1 above)
Comment (viktor-yengovatov) [reply src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java]: fixed

File: src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java (hunk 3 above)
Comment (viktor-yengovatov) [reply src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java]: done

File: src/com/intellij/idea/plugin/hybris/toolwindow/RemoteConnectionDialog.java (hunk 4 above)
Comment (viktor-yengovatov) [reply src/com/intellij/idea/plugin/hybris/toolwindow/RemoteConnectionDialog.java:44]: Done

File: src/com/intellij/idea/plugin/hybris/project/configurators/impl/DefaultLibRootsConfigurator.java (hunk 5 above)
Comment (viktor-yengovatov) [reply src/com/intellij/idea/plugin/hybris/project/configurators/impl/DefaultLibRootsConfigurator.java]: Removed. It should not have been commited at all

File: src/com/intellij/idea/plugin/hybris/toolwindow/RemoteConnectionDialog.form [hunk 6]
```diff
@@ -157,6 +157,26 @@
           <text resource-bundle="i18n/HybrisBundle" key="hybris.toolwindow.ssl.label"/>
         </properties>
       </component>
+      <component id="ef6c1" class="javax.swing.JComboBox" binding="sslProtocol">
+        <constraints>
+          <grid row="5" column="4" row-span="1" col-span="1" vsize-policy="0" hsize-policy="2" anchor="8" fill="1" indent="0" use-parent-layout="false"/>
+        </constraints>
+        <properties>
+          <model>
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/toolwindow/RemoteConnectionDialog.form:165]: Could we have a default value 'TLSv1' ?

update: I mean UI option.

Comment (viktor-yengovatov) [reply src/com/intellij/idea/plugin/hybris/toolwindow/RemoteConnectionDialog.form:165]: done

**Pull request merged**